js/lib/ValidateSPV.js

```diff
@@ -0,0 +1,231 @@
+      payload = btcUtils.extractOpReturnData(output);
+    } else {
+        let prefixHash = utils.safeSlice(output, 8, 10);
+        if (utils.typedArraysAreEqual(prefixHash, new Uint8Array([34, 0]))) {
+          // P2WSH
+          outputType = OUTPUT_TYPES.WSH;
+          payload = utils.safeSlice(output, 11, 43);
+        } else if (utils.typedArraysAreEqual(prefixHash, new Uint8Array([22, 0]))) {
+          // P2WPKH
+          outputType = OUTPUT_TYPES.WPKH;
+          payload = utils.safeSlice(output, 11, 31);
+        } else if (utils.typedArraysAreEqual(prefixHash, new Uint8Array([25, 118]))) {
+          // PKH
+          outputType = OUTPUT_TYPES.PKH;
+          payload = utils.safeSlice(output, 12, 32);
+        } else if (utils.typedArraysAreEqual(prefixHash, new Uint8Array([23, 169]))) {
+          // SH
+          outputType = OUTPUT_TYPES.SH;
+          payload = utils.safeSlice(output, 11, 31);
+        } else {
+          outputType = OUTPUT_TYPES.NONSTANDARD;
+          payload = null;
+        }
+    }
+
+    return { value, outputType, payload };
+  },
+
+  /**
+   * @notice                Parses a block header struct from a bytestring
+   * @dev                   Block headers are always 80 bytes, see Bitcoin docs
+   * @param {Uint8Array}    header Header
+   * @returns {object}      Header digest, version, previous block header hash, merkle root, timestamp, target, nonce
+   */
+  parseHeader: (header) => {
+    // If header has an invalid length, bubble up error
+    const EMPTY = utils.deserializeHex('0x0000000000000000000000000000000000000000000000000000000000000000');
+    if (header.length != 80) {
+      return { digest: EMPTY, version: 0n, prevHash: EMPTY, merkleRoot: EMPTY, timestamp: 0n, target: 0n, nonce: 0n };
+    }
+
+    let digest = btcUtils.reverseEndianness(btcUtils.hash256(header));
+    let version = utils.bytesToUint(btcUtils.reverseEndianness(utils.safeSlice(header, 0, 4)));
+    let prevHash = btcUtils.extractPrevBlockLE(header);
+    let merkleRoot = btcUtils.extractMerkleRootLE(header);
+    let timestamp = btcUtils.extractTimestamp(header);
+    let target = btcUtils.extractTarget(header);
+    let nonce = utils.bytesToUint(btcUtils.reverseEndianness(utils.safeSlice(header, 76, 80)));
+
+    return { digest, version, prevHash, merkleRoot, timestamp, target, nonce };
+  },
+
+  /**
+   * @notice                Checks validity of header chain
+   * @dev                   Compares the hash of each header to the prevHash in the next header
+   * @param {Uint8Array}    headers Raw byte array of header chain
+   * @returns {BigInt}      The total accumulated difficulty of the header chain, or an error code
+   */
+  validateHeaderChain: (headers) => {
+    // Check header chain length
+    if (headers.length % 80 !== 0) {
+      throw new Error('Header bytes not multiple of 80.');
+    }
+
+    // Initialize header start index
+    let digest;
+    let start = 0;
```

Comment:
`start` can be moved inside, but `digest` should remain outside because `validateHeaderPrevHash` uses the hash of the previous block.  If digest is moved inside the for loop, it will reset each time instead of storing the previous hash.

Reply:
fixed